massa-client/src/cmds.rs

```diff
@@ -613,6 +625,59 @@ impl Command {
                 }
                 Ok(Box::new(()))
             }
+            Command::send_smart_contract => {
+                if parameters.len() != 6 {
+                    bail!("wrong number of parameters");
+                }
+                let addr = parameters[0].parse::<Address>()?;
+                let path = parameters[1].parse::<PathBuf>()?;
+                let max_gas = parameters[2].parse::<u64>()?;
+                let gas_price = parameters[3].parse::<Amount>()?;
+                let coins = parameters[4].parse::<Amount>()?;
+                let fee = parameters[5].parse::<Amount>()?;
+
+                if !json {
+                    match gas_price
+                        .checked_mul_u64(max_gas)
+                        .and_then(|x| x.checked_add(coins))
+                        .and_then(|x| x.checked_add(fee))
+                    {
+                        Some(total) => {
+                            if let Ok(addresses_info) =
+                                client.public.get_addresses(vec![addr]).await
+                            {
+                                match addresses_info.get(0) {
+                                    Some(info) => {
+                                        if info.ledger_info.candidate_ledger_info.balance < total {
+                                            println!("WARNING: this operation may be rejected due to insuffisant balance");
+                                        }
+                                    }
+                                    None => println!("WARNING: address {} not found", addr),
```

Comment:
should we `bail` in the cases where it's sure to fail?

Reply:
We are not sure it will fail here. It may succeed if at the time we create the operation the address is not known of the network, but if it receives enough coins within its range of validity, it may be integrated.

Anyway the client does not guarantee that the operation is valid, only structurally valid. These warnings were added so people understand more easily why their transactions are never included.

Reply:
ok, thanks for the info.